client/client_test.go

```diff
@@ -215,3 +215,35 @@ func TestGetActionTimeout(t *testing.T) {
 		})
 	}
 }
+
+func Test_argListFromPgParams(t *testing.T) {
+	type testParams struct {
+		req      *agentpb.StartActionRequest_PTPgSummaryParams
+		expected []string
+	}
+
+	testCases := []*testParams{
+		{&agentpb.StartActionRequest_PTPgSummaryParams{Host: "10.20.30.40", Port: 555, Username: "person",
+			Password: "secret"}, []string{"--host", "10.20.30.40", "--port", "555", "--username", "person", "--password", "secret"}},
+		{&agentpb.StartActionRequest_PTPgSummaryParams{Host: "10.20.30.40", Port: 555, Username: "person",
+			Password: ""}, []string{"--host", "10.20.30.40", "--port", "555", "--username", "person"}},
+		{&agentpb.StartActionRequest_PTPgSummaryParams{Host: "10.20.30.40", Port: 555, Username: "",
+			Password: "secret"}, []string{"--host", "10.20.30.40", "--port", "555", "--password", "secret"}},
+		{&agentpb.StartActionRequest_PTPgSummaryParams{Host: "10.20.30.40", Port: 65536, Username: "",
+			Password: "secret"}, []string{"--host", "10.20.30.40", "--password", "secret"}},
+		{&agentpb.StartActionRequest_PTPgSummaryParams{Host: "", Port: 555, Username: "", Password: "secret"},
+			[]string{"--port", "555", "--password", "secret"}},
+
+		{&agentpb.StartActionRequest_PTPgSummaryParams{Host: "", Port: 0, Username: "", Password: ""}, []string{}},
+		{&agentpb.StartActionRequest_PTPgSummaryParams{Host: "", Port: 0, Username: "王华", Password: `"`},
```

Comment:
Can we add a test with a password having blank spaces?
I think in that case the password is not being quoted.